dist/common/scripts/scylla_util.py

```diff
@@ -105,6 +109,141 @@ def curl(url, byte=False):
+        return {"root": [root_dev], "ephemeral": [x for x in nvmes_present if not root_dev.startswith(os.path.join("/dev/", x))]}
+
+    def __populate_os_disks(self):
+        """populate disks from /dev/ and root mountpoint"""
+        __disks = {}
+        nvmes_present = self._non_root_nvmes()
+        for k, v in nvmes_present.items():
+            __disks[k] = v
+        return __disks
+
+    @staticmethod
+    def isNVME(gcpdiskobj):
+        """check if disk from GCP metadata is a NVME disk"""
+        if gcpdiskobj["interface"]=="NVME":
+            return True
+        return False
+
+    def __get_nvme_disks_from_metadata(self):
+        """get list of nvme disks from metadata server"""
+        import json
+        disksREST=self.__instance_metadata("disks")
+        disksobj=json.loads(disksREST)
+        nvmedisks=list(filter(self.isNVME, disksobj))
+        return nvmedisks
+
+    @property
+    def nvmeDiskCount(self):
+        """get # of nvme disks available for scylla raid"""
+        if self.__nvmeDiskCount is None:
+            try:
+                os_disks = self.__populate_os_disks()
+                ephemeral_disks = os_disks["ephemeral"]
+                count_os_disks=len(ephemeral_disks)
+            except Exception as e:
+                count_os_disks=0
+            nvme_metadata_disks = self.__get_nvme_disks_from_metadata()
+            count_metadata_nvme_disks=len(nvme_metadata_disks)
+            self.__nvmeDiskCount = count_os_disks if count_os_disks<count_metadata_nvme_disks else count_metadata_nvme_disks
+        return self.__nvmeDiskCount
+
+
+    @property
+    def instancetype(self):
+        """return the type of this instance"""
+        if self.__type is None:
+            self.__type = self.__instance_metadata("machine-type").split("/")[-1]
+        return self.__type
+
+    @property
+    def cpu(self):
+        """return the # of cpus of this instance"""
+        if self.__cpu is None:
+            self.__cpu = psutil.cpu_count()
+        return self.__cpu
+
+    @property
+    def memoryGB(self):
+        """return the size of memory in GB of this instance"""
+        if self.__memoryGB is None:
+            self.__memoryGB = psutil.virtual_memory().total/1024/1024/1024
+        return self.__memoryGB
+
+    def instance_size(self):
+        """Returns the size of the instance we are running in. i.e.: 2"""
+        return self.instancetype.split("-")[2]
+
+    def is_recommended_instance_size(self):
+        """if this instance has at least 2 cpus, it has a recommended size"""
+        if int(self.instance_size()) > 1:
+            return True
+        return False
+
+    # TODO below is in GB and is hardcoded, so this needs to be changed once the disks in GCP change
+    # note that they have 3TB physical devices actually, which they break into smaller 375GB disks and share the same mem with multiple machines
+    GCP_NVME_DISK_SIZE_2020=375
```

Comment:
This needs to be fixed.

Reply:
@avikivity  can you help me deciding what is better than this hardcode?

alternatives to hardcoded disk size is either a "seek" hack(which Vlad didn't like that much):
```
def get_file_size(filename):
    "Get the file size by seeking at end"
    fd= os.open(filename, os.O_RDONLY)
    try:
        return os.lseek(fd, 0, os.SEEK_END)
    finally:
        os.close(fd)
```

or a call to fdisk -l and parse from Disk info

or https://pypi.org/project/blkinfo/ (which would add dependency to code from pip)

Reply:
How about the stat() function?

Adding more dependencies should be avoided (better to add a dependency than to open-code, but in this case, stat() is fine).

Reply:
stat didn't work unfortunately for block devices

Reply:
I even tried sysfs ( disk_size= /sys/block/nvme0n1/queue/physical_block_size * /sys/block/nvme0n1/size ) and sysfs is buggy on GCP since it shows full size for the whole nvme card (so 3TB) )

Reply:
ok. I think seek() is fine then.